frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/UpdateAutomationModal.tsx

```diff
@@ -0,0 +1,319 @@
+/* Copyright Contributors to the Open Cluster Management project */
+
+import {
+    Cluster,
+    ClusterCurator,
+    ClusterCuratorDefinition,
+    createClusterCurator,
+    createResource,
+    IResource,
+    IRequestResult,
+    patchResource,
+    ResourceError,
+    ResourceErrorCode,
+    SecretDefinition,
+    ClusterCuratorKind,
+} from '../../../../../resources'
+import { makeStyles } from '@material-ui/styles'
+import {
+    AcmButton,
+    AcmForm,
+    AcmIcon,
+    AcmIconVariant,
+    AcmModal,
+    AcmSelect,
+    AcmTable,
+    IAcmTableColumn,
+} from '../../../../../ui-components'
+import {
+    ansibleCredentialsValue,
+    clusterCuratorSupportedCurationsValue,
+    validClusterCuratorTemplatesValue,
+} from '../../../../../selectors'
+import { clusterCuratorsState } from '../../../../../atoms'
+import {
+    Button,
+    ButtonVariant,
+    Flex,
+    FlexItem,
+    ModalVariant,
+    SelectOption,
+    Stack,
+    StackItem,
+} from '@patternfly/react-core'
+import { useMemo, useState } from 'react'
+import { useTranslation } from '../../../../../lib/acm-i18next'
+import { useRecoilValue } from 'recoil'
+import { Link } from 'react-router-dom'
+import { useClusterDistributionColumn, useClusterProviderColumn } from '../ManagedClusters'
+import { NavigationPath } from '../../../../../NavigationPath'
+import { cloneDeep } from 'lodash'
+
+const useStyles = makeStyles({
+    body: {},
+    select: {
+        '& > div': {
+            alignItems: 'end',
+            '& > div:last-child': {
+                paddingBottom: '6px',
+            },
+        },
+    },
+    table: {
+        '& .pf-c-toolbar': {
+            display: 'none',
+        },
+    },
+})
+
+export function UpdateAutomationModal(props: {
+    close: () => void
+    open: boolean
+    clusters: Cluster[] | undefined
+}): JSX.Element {
+    const { t } = useTranslation()
+    const classes = useStyles()
+    const validCuratorTemplates = useRecoilValue(validClusterCuratorTemplatesValue)
+    const clusterCurators = useRecoilValue(clusterCuratorsState)
+    const [selectedCuratorTemplate, setSelectedCuratorTemplate] = useState<ClusterCurator | undefined>()
+    const [isUpdating, setIsUpdating] = useState(false)
+    const clusterProviders = useClusterProviderColumn()
+    const distributionVersion = useClusterDistributionColumn(clusterCurators)
+    const supportedCurations = useRecoilValue(clusterCuratorSupportedCurationsValue)
+    const ansibleCredentials = useRecoilValue(ansibleCredentialsValue)
+
+    const handleCuratorSelect = (uid: string | undefined) => {
+        setSelectedCuratorTemplate(
+            !uid ? undefined : validCuratorTemplates.find((template) => template.metadata.uid === uid)
+        )
+    }
+
+    const isupdatable = (cluster: Cluster) => !cluster.distribution?.isManagedOpenShift
```

Comment:
Still need to check that cluster:

- is OCP
- is Ready
- is not currently upgrading

See https://github.com/stolostron/console/blob/main/frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/BatchUpgradeModal.tsx#L35 for example.